runtime/packages/core-modules/assembly/executive.ts

```diff
@@ -47,6 +57,94 @@ export namespace Executive{
     export function finalizeBlock(): Header {
         return System.finalize();
     }
+    /**
+     * creates inherents from internal modules
+     * @param data inherents
+     */
+    export function createExtrinsics(data: InherentData): u8[] {
+        const timestamp: Inherent = Timestamp.createInherent(data);
+        return System.ALL_MODULES.concat(timestamp.toU8a());
+    }
+
+    /**
+     * Apply Extrinsics
+     * @param ext extrinsic
+     */
+    export function applyExtrinsic(ext: u8[]): u8[] {
+        const encodedLen = Bytes.decodeCompactInt(ext);
+        return Executive.applyExtrinsicWithLen(ext, encodedLen.value as u32);
+    }
+    
+    /**
+     * 
+     * @param ext extrinsic
+     * @param encodedLen length
+     * @param encoded encoded extrinsic
+     */
+    export function applyExtrinsicWithLen(ext: u8[], encodedLen: u32): u8[]{
+        // if the length of the encoded extrinsic is less than 144, it's inherent
+        // in our case, we have only timestamp inherent
+        if(Inherent.isInherent(ext)){
+            const inherent = Inherent.fromU8Array(ext).result;
+            Timestamp.set(inherent.arg.value);
+            Timestamp.toggleUpdate();
+            return ApplyExtrinsicResult.SUCCESS;
+        }
+        const cmpLen = Bytes.decodeCompactInt(ext);
+        ext = ext.slice(cmpLen.decBytes);
+        const result = ext.shift();
+        if (result as bool){
+            const extrinsic = Extrinsic.fromU8Array(ext).result;
+            const sender: AccountId = AccountId.fromU8Array(extrinsic.from.toU8a()).result;
+            const receiver: AccountId = AccountId.fromU8Array(extrinsic.to.toU8a()).result;
+            BalancesModule.transfer(sender, receiver, extrinsic.amount.value);
+            return ApplyExtrinsicResult.SUCCESS;
+        }
+        return ApplyExtrinsicResult.CALL_ERROR;
+    }
+
+    /**
+     * Initial transaction validation
+     * @param source source of the transaction (external, inblock, etc.)
+     * @param utx transaction
+     */
+    export function validateTransaction(source: u8[], utx: Extrinsic): u8[] {
+        const from: AccountId = AccountId.fromU8Array(utx.from.toU8a()).result;
+        const fromBalance = BalancesModule.getAccountData(from);
+        const transfer = utx.getTransferBytes();
+
+        if(!System.verifySignature(utx.signature, transfer, from)){
+            Log.error("Validation error: Invalid signature");
+            return TransactionError.BAD_PROOF_ERROR;
+        }   
+        const nonce = System.accountNonce(from);
+        if (<u64>nonce.value >= <u64>utx.nonce.value){
+            Log.error("Validation error: Nonce value is less than or equal to the latest nonce");
+            return TransactionError.STALE_ERROR;
```

Comment:
I think that a better error name would be `NONCE_TOO_LOW` as this is standard error in Ethereum